src/main/java/com/clanki/commands/UpdateCommand.java

```diff
@@ -57,13 +57,22 @@ public int implementUpdate(ArrayList<Flashcard> flashcards, String userText) {
     public void execute(FlashcardList flashcardList, Ui display) {
         ArrayList<Flashcard> flashcards = flashcardList.getFlashCards();
         findFlashcard(flashcards, query);
-        System.out.println(
-                "Found " + matchingFlashcards.size() + " cards with the query \"" + query + "\":");
-        printFlashCards(matchingFlashcards);
-        System.out.println("Which flashcard do you want to update?");
-        String userText = display.getUserCommand();
-        int index = implementUpdate(flashcards, userText);
-        System.out.println("Understood. The card has been updated to");
-        printFlashCard(flashcards.get(index));
+        if (matchingFlashcards.size() > 0) {
+            System.out.println(
+                    "Found " + matchingFlashcards.size() + " cards with the query \"" + query + "\":");
+            printFlashCards(matchingFlashcards);
+            System.out.println("Which flashcard do you want to update?");
+            String userText = display.getUserCommand();
+            int index = 0;
+            try {
+                index = implementUpdate(flashcards, userText);
```

Comment:
You may want to consider the case where the user only inputs an integer as their update input, and what the program should do about it